spacy/tests/test_language.py

```diff
@@ -528,3 +528,28 @@ def test_language_source_and_vectors(nlp2):
     assert long_string in nlp2.vocab.strings
     # vectors should remain unmodified
     assert nlp.vocab.vectors.to_bytes() == vectors_bytes
+
+
+@pytest.mark.parametrize("n_process", [1, 2])
+def test_pass_doc_to_pipeline(nlp, n_process):
+    texts = ["cats", "dogs", "guinea pigs"]
+    docs = [nlp.make_doc(text) for text in texts]
+    assert not any(len(doc.cats) for doc in docs)
+    doc = nlp(docs[0])
+    assert doc.text == texts[0]
+    assert len(doc.cats) > 0
+    docs = nlp.pipe(docs, n_process=n_process)
+    assert [doc.text for doc in docs] == texts
+    assert all(len(doc.cats) for doc in docs)
```

Comment:
This test needs to check for CPU ops only or it will hang/crash.

Reply:
I'll get that check in, but these should really be xfailing tests rather than hardcoding these kind of exceptions :|

Reply:
(but for another day...)

Reply:
This is how I've done it so far, if you know of a more general way to handle this, then that would be welcome, but for another day...